ruma-federation-api/src/serde/pdu_process_response.rs

```diff
@@ -0,0 +1,68 @@
+use std::{collections::BTreeMap, fmt};
+
+use ruma_identifiers::EventId;
+use serde::{
+    de::{Deserializer, MapAccess, Visitor},
+    ser::{SerializeMap, Serializer},
+    Deserialize, Serialize,
+};
+
+pub fn serialize<S>(
+    response: &BTreeMap<EventId, Result<(), String>>,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let mut map = serializer.serialize_map(Some(response.len()))?;
+    for (key, value) in response {
+        let wrapped_error = WrappedError {
+            error: match value {
+                Ok(_) => None,
```

Comment:
I think serializing `None` to JSON is going to result in `null`, is that what we want here?

Reply:
Added the skip_serializing_if attribute to fix it and added tests to verify.